symphony/bdk/core/service/signal/signal_service.py

```diff
@@ -0,0 +1,177 @@
+from symphony.bdk.core.auth.auth_session import AuthSession
+from symphony.bdk.gen.agent_api.signals_api import SignalsApi
+from symphony.bdk.gen.agent_model.base_signal import BaseSignal
+from symphony.bdk.gen.agent_model.channel_subscriber import ChannelSubscriber
+from symphony.bdk.gen.agent_model.channel_subscription_response import ChannelSubscriptionResponse
+from symphony.bdk.gen.agent_model.signal import Signal
+from symphony.bdk.gen.agent_model.signal_list import SignalList
+
+
+class OboSignalService:
+    """Service class exposing OBO-enabled endpoints to manage signal information.
+
+    This service is used for listing signals related to the OBO user, get information of a specified signal
+    or perform some actions related to the signal like:
+
+    * List signals
+    * Get a signal
+    * Create a signal
+    * Update a signal
+    * Delete a signal
+    * Subscribe or unsubscribe a signal
+    """
+
+    def __init__(self, signals_api: SignalsApi, auth_session: AuthSession):
+        self._signals_api = signals_api
+        self._auth_session = auth_session
+
+    async def list_signals(self, skip: int = 0, limit: int = 50) -> SignalList:
+        """ Lists signals on behalf of the user. The response includes signals that the user has created and
+        public signals to which they have subscribed.
+
+        See: 'List signals <https://developers.symphony.com/restapi/reference#list-signals>'_
+
+        :param skip: The number of signals to skip.
+        :param limit: Maximum number of signals to return. Default is 50, maximum value is 500.
+        :return: List of signals found.
+        """
+        params = {
+            'skip': skip,
+            'limit': limit,
+            'session_token': await self._auth_session.session_token,
+            'key_manager_token': await self._auth_session.key_manager_token
+        }
+        return await self._signals_api.v1_signals_list_get(**params)
```

Comment:
IMHO I'd prefer to have parameters directly unpacked like `self._signals_api.v1_signals_list_get(skip=skip, limit=limit, ...)`.
Having an intermediary `params` dict is only useful when you have a dynamic number of params.

Reply:
Done in the next commit :)